### ISSUE TITLE
toIdentifier(statuses.message[code]) causes "Cannot read property '100' of undefined"

### ISSUE THREAD

**Issue by caugner** [jshttp/http-errors#88]
[We are getting](https://github.com/mdn/yari/pull/5777#issuecomment-1078469292) the following error after bumping send from 0.17.2 to 0.18.0, which bumped http-errors from 1.8.1 to 2.0.0:

```
/home/runner/work/yari/yari/mdn/content/node_modules/@mdn/yari/node_modules/send/node_modules/http-errors/index.js:261
    var name = toIdentifier(statuses.message[code])
                                            ^

TypeError: Cannot read property '100' of undefined
    at forEachCode (/home/runner/work/yari/yari/mdn/content/node_modules/@mdn/yari/node_modules/send/node_modules/http-errors/index.js:261:45)
    at Array.forEach (<anonymous>)
    at populateConstructorExports (/home/runner/work/yari/yari/mdn/content/node_modules/@mdn/yari/node_modules/send/node_modules/http-errors/index.js:259:9)
    at Object.<anonymous> (/home/runner/work/yari/yari/mdn/content/node_modules/@mdn/yari/node_modules/send/node_modules/http-errors/index.js:31:1)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
error Command failed with exit code 1.
```

Line:

https://github.com/jshttp/http-errors/blame/206aa2c15635dc1212c06c279540972aa90e23ea/index.js#L261

Commit: 8c38486f4366002087a30f939f69016dc2a385c3

**Comment by dougwilson** [comment]
Hi @caugner that is very strange. I'm not able to reproduce the error, and neither the CI of send nor this module is.

Can you provide a minimal reproducable example thay demonstrates the issue so I can debug it? Alternatively you are always welcome to make a PR that adds whatever test is missing that is not catching this and a fix for it 👍

**Comment by caugner** [comment]
Point taken.

FWIW My best guess it's related to having a different version of `statuses` hoisted:

```console
$ yarn why statuses
yarn why v1.22.17
...
=> Found "statuses@1.5.0"
info Has been hoisted to "statuses"
info Reasons this module exists
   - Hoisted from "http-errors#statuses"
   - Hoisted from "express#statuses"
   - Hoisted from "express#finalhandler#statuses"
   - Hoisted from "express#send#statuses"
   - Hoisted from "express#serve-static#send#statuses"
   - Hoisted from "react-scripts#webpack-dev-server#serve-index#http-errors#statuses"
...
=> Found "send#statuses@2.0.1"
info Reasons this module exists
   - "send" depends on it
   - Hoisted from "send#http-errors#statuses"
...
✨  Done in 0.68s.
```

This really shouldn't cause any trouble, and it might just be that our build process is doing something weird.

I will close this issue, as I'm unable to provide a minimal reproducable example, and because `http-errors` is most likely not even the culprit here.

**Comment by domfie** [comment]
Had the same problem updating strapi. I deleted the whole node_modules folder and did a fresh install and everything worked again 🙌

**Comment by Zambonilli** [comment]
I hit this error as well with express but am also not sure how to reproduce. My module was referencing ^1.8.0 and express ^4.17.0 in it's package.json. However, express 4.18.x upgraded to http-errors 2.x from 1.8.x. What's weird is that yarn had ^1.8.0 & 2.0.0 resolved to the same http-errors 2.0.0 in my lock file. So, when booting I would receive the cannot read property 100 error. I was able to resolve by pinning express to 4.17.3 and http-errors to 1.8.1 explicitly in my package.json, cleaning the yarn cache, deleting node_modules and doing a new yarn install.
